src/EventSource.js

```diff
@@ -115,6 +116,7 @@ class EventSource {
           this._handleEvent(xhr.responseText || '');
 
           if (xhr.readyState === XMLHttpRequest.DONE) {
+            this.dispatch('done', { type: 'done' });
```

Comment:
I would strongly recommend to dispatch the `done` event _after_ calling `_pollAgain`. This would allow users to cancel the reconnection by calling `.close()` from the `done` handler.

Reply:
Good call. I just updated and tested that it works as expected.